File: Cap 12/12.4 - Metodo Interno/cometint.dart
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.4
 * 
 * Programa ..: USO DE METODO INTERNO A UMA CLASSE
 * 
 */
 
import 'dart:io';

// Definicao de classe com metodo interno

class CLS_ALUNO {

  late String NOME;
  late List<double> NOTAS = List<double>.filled(4, 0);
  late double MEDIA;

  double CMEDIA() {
    double SOMA = 0;
    for (int I = 0; I <= 3; I++) {
      SOMA = SOMA + NOTAS[I];
    }
    MEDIA = SOMA / 4;
    return MEDIA;
  }
  
}

void main() {
  CLS_ALUNO ALUNO = CLS_ALUNO();

  print("DADOS ESCOLARES");
  print("");
  
  stdout.write("Informe o nome: ");
  ALUNO.NOME = stdin.readLineSync()!;

  print("Informe as notas:");
  for (int I = 0; I <= 3; I++) {
    stdout.write("${I + 1}a. nota: ");
    ALUNO.NOTAS[I] = double.parse(stdin.readLineSync()!);
  }
  
  ALUNO.CMEDIA();

  print("");
  print("RELATORIO ESCOLAR");
  print("");
  
  print("Nome: ${ALUNO.NOME}");
  for (int I = 0; I <= 3; I++) {
     print("${I + 1}a. nota: ${ALUNO.NOTAS[I].toStringAsFixed(1).padLeft(5)}");
  }
  print("Media: ${ALUNO.MEDIA.toStringAsFixed(1).padLeft(5)}");
  
}

File: Cap 12/12.4 - Metodo Interno/cometint.rb
=begin
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.4
 * 
 * Programa ..: USO DE METODO INTERNO A UMA CLASSE
 * 
=end

class CLS_ALUNO
  attr_accessor :nome, :notas, :media

  def cmedia()
    soma = 0
    for i in 0...4
      soma += notas[i]
    end
    self.media = soma / 4.0
    return media
  end
end

aluno = CLS_ALUNO.new

puts "DADOS ESCOLARES"
puts

print "Informe o nome: "
aluno.nome = gets.chomp

puts "Informe as notas: "
aluno.notas = []

for i in 0...4
  print "#{i + 1}a. nota: "
  aluno.notas << gets.chomp.to_f
end

aluno.cmedia()

puts
puts "RELATORIO ESCOLAR"
puts
puts "Nome: #{aluno.nome}"
for i in 0...4
  puts "#{i + 1}a. nota: #{aluno.notas[i]}"
end
puts "Media: #{aluno.media}"
